### ISSUE TITLE
src/block_base.h:15: error: redefinition of typedef ‘BlockBase’ by GCC 4.4.7

### ISSUE THREAD

**Issue by smtakeda** [Legrandin/pycryptodome#3]
In Centos6.5, GCC 4.4.7, Python 2.7.10, pip install fails:
```
pip install pycryptodome
```
and got this error
```
...
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DLTC_NO_ASM -DHAVE_CPUID_H -Isrc/ -I/usr/local/include/python2.7 -c src/AES.c -o build/temp.linux-x86_64-2.7/src/AES.o
In file included from src/AES.c:29:
src/block_base.h:15: error: redefinition of typedef ‘BlockBase’
src/block_base.h:6: note: previous declaration of ‘BlockBase’ was here
error: command 'gcc' failed with exit status 1
```
If I switch CC to gcc4.7, it works.
```
CC=gcc47 pip install pycryptodome
```
NOTE: here is my virtualenv:
```
> pip list
pip (7.1.2)
setuptools (18.4)
wheel (0.26.0)
```
Is there any a supported/tested minimum version of GCC? There seems some GCC behavior changes between 4.4 and 4.7.

**Comment by smtakeda** [comment]
This stackoverflow talks about the issue: http://stackoverflow.com/questions/6526322/why-redefinition-of-typedef-error-with-gcc-4-3-but-not-gcc-4-6

**Comment by Legrandin** [comment]
Thanks for reporting this. I have fixed this problem with 534e105e9539477ce4d2eb6120e1a01d53736251. I hope to push out a release including this fix this week.

**Comment by smtakeda** [comment]
Thanks!
